### PR TITLE
Extract PrivateAssets class from 18F/hub

### PR DESCRIPTION
**Pull request by mbland**
This also makes use of my new [test_temp_file_helper gem](https://github.com/18F/test_temp_file_helper).

cc: @afeld @gboone @adelevie

### CODE REVIEW

File: lib/team_hub/private_assets.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    # Copies a directory of private data assets into the generated site.
+    # @param site [Jekyll::Site] Jekyll site object
+    # @param dir_to_copy [string] directory within
+    #   site.config['private_data_path'] to copy into the generated site
+    def self.copy_directory_to_site(site, dir_to_copy)
+      private_root = File.join(site.source, site.config['private_data_path'])
+      source_dir = File.join(private_root, dir_to_copy)
+      return unless Dir.exists? source_dir
+      d = Dir.open(source_dir)
```
Comment (afeld) [inline lib/team_hub/private_assets.rb]: You'll want to [`#close`](http://www.ruby-doc.org/core-2.2.0/Dir.html#method-i-close) the `Dir` instance, or better yet, use the block form of [`.open`](http://www.ruby-doc.org/core-2.2.0/Dir.html#method-c-open).

File: test/private_assets_copy_directory_to_site_test.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+require_relative "site"
+
+require "minitest/autorun"
+require "test_temp_file_helper"
+
+module TeamHub
+  class PrivateAssetsCopyDirectoryToSiteTest < ::Minitest::Test
+    def setup
+      config = {
+        'source' => ENV['TEST_TMPDIR'],
```
Comment (afeld) [inline test/private_assets_copy_directory_to_site_test.rb]: I believe that using code in one place to set an environment variable to be used in other parts of the same application is an anti-pattern. What if this value were retrieved via a method like `TestTempFileHelper.test_tmpdir`?

File: lib/team_hub/private_assets.rb (hunk 1 above)
Comment (mbland) [inline lib/team_hub/private_assets.rb]: Done.

File: test/private_assets_copy_directory_to_site_test.rb (hunk 2 above)
Comment (mbland) [inline test/private_assets_copy_directory_to_site_test.rb]: Done after update to `test_temp_file_helper` v0.0.2.

File: Gemfile.lock [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+PLATFORMS
+  ruby
+
+DEPENDENCIES
+  bundler (~> 1.7)
+  codeclimate-test-reporter
+  minitest
+  rake (~> 10.0)
+  team_hub!
+  test_temp_file_helper
```
Comment (afeld) [inline Gemfile.lock]: Mind removing+gitignoring the Gemfile.lock?

http://stackoverflow.com/a/4151540/358804

**Comment by afeld** [comment]
Just the one comment. Otherwise, :shipit:

**Pull request merged**